setup.md

```diff
@@ -269,8 +269,124 @@ TODO: Explain one of the following:
+
+### Installing readline.lua
+For the official support on getting readline.lua, see the
+[official docs](https://pjb.com.au/comp/lua/readline.html#installation).
+
+The easiest way to get readline.lua is to install it with Luarocks, which
+will fetch the package and automatically compile the native bindings for you.
+
+#### Installing readline with LuaRocks
+
+```bash
+# to install globally on the system (requires admin privileges or sudo)
+$ luarocks install readline
+
+# to install to the user's local tree
+$ luarocks install --local readline
+
+# install for Lua 5.1 (including LuaJIT)
+$ luarocks install --lua-version=5.1 readline
+```
+
+Because the readline Lua module contains native bindings to libreadline, be sure
+it's installed for the Lua version you intend to use.
+
+If you've installed with the `--local` flag, you may need to ensure your `package.path`
+and `package.cpath` contain its location so it can be resolved by `require`.
+
+**TODO: Add and link to section on configuring environment to detect local rocks**
+
+#### Manual installation
+
+See the [installation guide in the readline.lua docs](https://pjb.com.au/comp/lua/readline.html#installation)
+for a URL from which to obtain the source code of the most recent version.
+The source is also available, along with the author's other Lua modules, on
+[Gitlab](https://gitlab.com/peterbillam/pjb_lua).
+
+To manually compile the native bindings, you can use `gcc`. Here is an example
+on a typical Linux system:
+
+```bash
+$ gcc -O2 -fPIC -c C-readline.c -o C-readline.o
+$ gcc -shared -o C-readline.so C-readline.o -lreadline
+
+# to compile for a specific Lua version, you will need to include its location:
+$ gcc -O2 -fPIC -I/usr/include/luajit-2.0 -c C-readline.c -o C-readline.o
+$ gcc -shared -o C-readline.so C-readline.o -lreadline
+```
+
+Once the bindings are compiled, simply ensure readline.lua is available on your
+`package.path` and C-readline.so on `package.cpath`.
+
+### Configuring readline for an enhanced experience
+
+Readline itself has a number of configuration options, which can be set either
+via the readline.lua API in `fennelrc`, or in readline's own `~/.inputrc` config file.
+
+#### Enabling persistent history
+
+To configure the REPL to save the rolling history to file at the end of every
+session, add the following to your `fennelrc` with your desired filename:
+
+See the readline.lua documentation for information on its API, most notably
+other parameters that can be set via
+[rl.set_options](https://pjb.com.au/comp/lua/readline.html#set_options).
+
+```fennel
+; persist repl history
+(match package.loaded.readline
+  rl   (rl.set_options {:histfile  "~/.fennel_history" ; default:"" (don't save)
+                        :keeplines 1000}))             ; default:1000
+```
+
+#### Configuring readline in `~/.inputrc`
+
+[documentation on the readline init file](https://www.gnu.org/software/bash/manual/html_node/Readline-Init-File.html)
```

Comment:
Let's make this a complete sentence.

Reply:
... not sure how i let that happen :clown_face: